crates/bevy_render/src/mesh/shape/icosphere.rs

```diff
@@ -20,8 +20,31 @@ impl Default for Icosphere {
     }
 }
 
-impl From<Icosphere> for Mesh {
-    fn from(sphere: Icosphere) -> Self {
+#[derive(Debug, Clone)]
+pub enum FromIcosphereError {
+    TooManyVertices {
+        subdivisions: usize,
+        number_of_resulting_points: usize,
+    },
+}
+
+impl std::error::Error for FromIcosphereError {}
+
+impl std::fmt::Display for FromIcosphereError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FromIcosphereError::TooManyVertices {
+                subdivisions,
+                number_of_resulting_points,
+            } => write!(f, "Cannot create an icosphere of {subdivisions} subdivisions due to there being too many vertices being generated: {number_of_resulting_points}. (Limited to 65535 vertices or 79 subdivisions)")
```

Comment:
Nit: This crate comes with `thiserror` I believe, so you could also just use that to generate the `Error` and `Display` impls

Reply:
Wasn't quite sure if we were using thiserror extensively
So I decided to make an Error impl instead.
I can convert it to use thiserror instead if it's okay

Reply:
I think we should just be consistent. If we don't want to use it, we should probably remove it. If we do want to use it, we should prefer its usage when possible